src/uct/ib/mlx5/gga/gga_mlx5.c

```diff
@@ -272,9 +273,11 @@ uct_gga_mlx5_rkey_resolve(uct_ib_mlx5_md_t *md,
         return UCS_OK;
     }
 
+    pthread_mutex_lock(&mem_attach_lock);
```

Comment:
why need lock - which data struct do we protect here?

Reply:
added the comment